test/nodes/test_prompt_node.py

```diff
@@ -11,10 +11,32 @@
 from haystack.nodes.prompt.prompt_node import HFLocalInvocationLayer
 
 
-def is_openai_api_key_set(api_key: str):
+def is_api_key_set(api_key: str):
     return len(api_key) > 0 and api_key != "KEY_NOT_FOUND"
 
 
+def skip_test_for_invalid_key(prompt_model):
+    if prompt_model.api_key is not None and not is_api_key_set(prompt_model.api_key):
+        pytest.skip("No API key found, skipping test")
+
+
+def get_either_openai_or_azure_openai_api_key():
+    openai_api_key = os.environ.get("OPENAI_API_KEY", None)
+    azure_openai_api_key = os.environ.get("AZURE_OPENAI_API_KEY", None)
+    if azure_openai_api_key:
+        return azure_openai_api_key
```

Comment:
I think if both API keys are present in the env then only the Azure API will be tested right, since that is the first if statement that is hit. 

Instead of the `@either_openai_or_azure_openai_api_key_in_env` approach it would be more straightforward to label tests as either azure, openai or both as you've done in other tests when using this label 
```python
@pytest.mark.parametrize("prompt_model", ["hf", "openai", "azure"], indirect=True)
```

Reply:
Cool, will do

Reply:
I think it would be best if we removed this fixture altogether and only use ```python
@pytest.mark.parametrize("prompt_model", ["hf", "openai", "azure"], indirect=True)
```